spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -163,8 +167,25 @@ public DistributionMode distributionMode() {
     return DistributionMode.fromName(modeName);
   }
 
-  public DistributionMode deleteDistributionMode() {
-    return rowLevelCommandDistributionMode(TableProperties.DELETE_DISTRIBUTION_MODE);
+  public DistributionMode copyOnWriteDeleteDistributionMode() {
+    String deleteModeName = confParser.stringConf()
+        .option(SparkWriteOptions.DISTRIBUTION_MODE)
+        .tableProperty(TableProperties.DELETE_DISTRIBUTION_MODE)
+        .parseOptional();
+
+    if (deleteModeName != null) {
+      // range distribution only makes sense if the sort order is set
+      DistributionMode deleteMode = DistributionMode.fromName(deleteModeName);
+      if (deleteMode == RANGE && table.sortOrder().isUnsorted()) {
+        return HASH;
+      } else {
+        return deleteMode;
+      }
+    } else {
+      // use hash distribution if write distribution is range or hash
```

Comment:
Why not inherit the range mode? If we support it explicitly anyway then it makes sense to use it right?

Reply:
One reason is to avoid changing the behavior we have right now. The second reason is performance. I think it is pretty nice that we can do a hash partitioning by file as it is way more efficient than a range-based shuffle.

Reply:
Added a comment below.

Reply:
I agree with this one. No range unless delete mode is range.